src/sync/sync_manager.cpp

```diff
@@ -32,6 +35,73 @@ SyncManager& SyncManager::shared()
     return manager;
 }
 
+void SyncManager::configure_file_system(const std::string& base_file_path,
+                                        MetadataMode metadata_mode,
+                                        util::Optional<std::vector<char>> custom_encryption_key)
+{
+    std::vector<std::tuple<std::string, std::shared_ptr<SyncUser>>> users_to_add;
+    REALM_ASSERT(!m_file_manager && !m_metadata_manager);   // Can only call this once
+    m_file_manager = std::make_unique<SyncFileManager>(base_file_path);
+    switch (metadata_mode) {
+        case MetadataMode::NoEncryption:
+            m_metadata_manager = std::make_unique<SyncMetadataManager>(m_file_manager->metadata_path(),
+                                                                       false);
+            break;
+        case MetadataMode::Encryption:
+            m_metadata_manager = std::make_unique<SyncMetadataManager>(m_file_manager->metadata_path(),
+                                                                       true,
+                                                                       std::move(custom_encryption_key));
+            break;
+        case MetadataMode::NoMetadata:
+            return;
+    }
+
+    REALM_ASSERT(m_metadata_manager);
+    // Load persisted users into the users map.
+    SyncUserMetadataResults users = m_metadata_manager->all_unmarked_users();
+    for (size_t i = 0; i < users.size(); i++) {
```

Comment:
My custom collection type doesn't support it yet. I'll add it as a followup item.